app/src/main/java/com/github/se/studybuddies/calender/CalendarScreen.kt

```diff
@@ -0,0 +1,189 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.KeyboardArrowLeft
+import androidx.compose.material.icons.filled.KeyboardArrowRight
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.studybuddies.R
+import com.github.se.studybuddies.data.CalendarUiState
+import com.github.se.studybuddies.navigation.NavigationActions
+import com.github.se.studybuddies.navigation.Route
+import com.github.se.studybuddies.ui.shared_elements.GoBackRouteButton
+import com.github.se.studybuddies.ui.shared_elements.Sub_title
+import com.github.se.studybuddies.ui.shared_elements.TopNavigationBar
+import com.github.se.studybuddies.ui.theme.Blue
+import com.github.se.studybuddies.ui.theme.LightBlue
+import com.github.se.studybuddies.viewModels.CalendarViewModel
+import java.time.YearMonth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CalendarApp(
+    viewModel: CalendarViewModel = androidx.lifecycle.viewmodel.compose.viewModel(),
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("calendar_scaffold"),
+      topBar = {
+        TopNavigationBar(
+            title = { Sub_title(title = "Calendar") },
+            navigationIcon = {
+              GoBackRouteButton(navigationActions = navigationActions, Route.SOLOSTUDYHOME)
+            },
+            actions = {})
+      },
+  ) { padding ->
+    Surface(
+        modifier = Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(padding)) {
+          CalendarWidget(
+              days = DateUtil.daysOfWeek,
+              yearMonth = uiState.yearMonth,
+              dates = uiState.dates,
+              onPreviousMonthButtonClicked = { prevMonth -> viewModel.toPreviousMonth(prevMonth) },
+              onNextMonthButtonClicked = { nextMonth -> viewModel.toNextMonth(nextMonth) },
+              onDateClickListener = {
```

Comment:
Ok I think a bit and I think they could probably be a better way to initialize CalendarWidget function. The parameters yearsMonth,dates, onPreviousMonthButtonClicked, onNextButtonClicked (and maybe onDateClickListener), they all need/are initialize by the viewModel. Then instead of passing all these different parameters inside of the CalendarWidget, I will just pass the viewModel as a parameter. Then you could initialize the other parameters later. 
Also I realized that you aren't using yearMonth, dates,onPreviousMonthButtonClicked and onNextMonthButtonClicked on the CalendarWidget. You're only taking these parameters to give them to Header and Content. Thus I will as well just give the viewModel to Header and Contente instead of all the other parameters. Therere yearMonth, dates,onPreviousMonthButtonClicked and onNextMonthButtonClicked will only be created when needed in the appropriate function at the right timing. 
What do you think of it ?

Reply:
Hi, thanks you for your comment.  it is just a way of coding , instead of instantiating the variable in the CalendarWidget I do it in CalendarApp so that if I want to reuse calendar Widget for other function I can, for exemple if I code the shared calendar I can reuses this function in the screen of the shared variable , it is just for modularity because I reuse a lot of the function in other classes , and I am passing the parameter of the view model prevMonth -> viewModel.toPreviousMonth(prevMonth)